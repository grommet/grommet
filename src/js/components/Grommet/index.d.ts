import * as React from 'react';
import { ThemeType } from '../../themes';
import { BackgroundType } from '../../utils';

export interface GrommetProps {
  background?: BackgroundType;
  containerTarget?: HTMLElement;
  cssVars?: boolean;
  dir?: 'rtl';
  full?: boolean | 'min';
  options?: {
    box?: {
      cssGap?: boolean;
    };
    layer?: {
      singleId?: boolean;
    };
    drop?: {
      checkContainingBlock?: boolean;
    };
  };
  messages?: {
    messages?: {
      button?: {
        busy?: string;
        success?: string;
      };
      calendar?: {
        previousMove?: string;
        previous?: string;
        nextMove?: string;
        next?: string;
      };
      carousel?: {
        previous?: string;
        next?: string;
        jump?: string;
      };
      dateInput?: {
        openCalendar?: string;
        enterCalendar?: string;
        exitCalendar?: string;
      };
      dataChart?: {
        detailTitle?: string;
        detailFocus?: string;
      };
      dataFilters?: {
        clear?: string;
        heading?: string;
        open?: string;
        openSet?: {
          singular?: string;
          plural?: string;
        };
      };
      dataForm?: {
        submit?: string;
      };
      dataSearch?: {
        label?: string;
        open?: string;
      };
      dataSort?: {
        ascending?: string;
        by?: string;
        descending?: string;
        direction?: string;
        open?: string;
      };
      dataSummary?: {
        filtered?: string;
        filteredSingle?: string;
        items?: string;
        itemsSingle?: string;
        selected?: string;
        total?: string;
        totalSingle?: string;
      };
      dataTable?: {
        ascending?: string;
        collapse?: string;
        collapseAll?: string;
        decrease?: string;
        descending?: string;
        increase?: string;
        expand?: string;
        expandAll?: string;
        resizerAria?: string;
        rows?: string;
        rowsChanged?: string;
        rowsSingle?: string;
        searchBy?: string;
        total?: string;
        totalSingle?: string;
      };
      dataTableColumns?: {
        open?: string;
        order?: string;
        pinned?: string;
        select?: string;
        tip?: string;
        orderAria?: string;
        selectAria?: string;
      };
      dataTableGroupBy?: {
        clear?: string;
        label?: string;
      };
      dataView?: {
        label?: string;
      };
      fileInput?: {
        alert?: {
          maxFile?: string;
          maxSize?: string;
        };
        browse?: string;
        dropPrompt?: string;
        dropPromptMultiple?: string;
        files?: string;
        maxFile?: string;
        maxSizeSingle?: string;
        maxSizeMultiple?: {
          singular?: string;
          plural?: string;
        };
        remove?: string;
        removeAll?: string;
      };
      form?: {
        invalid?: string;
        required?: string;
      };
      formField?: {
        maxCharacters?: {
          remaining?: {
            singular?: string;
            plural?: string;
          };
          overLimit?: {
            singular?: string;
            plural: string;
          };
        };
      };
      list?: {
        pinned?: string;
      };
      menu?: {
        openMenu?: string;
        closeMenu?: string;
      };
      meter?: {
        bar?: {
          singular?: string;
          plural?: string;
        };
        circle?: {
          singular?: string;
          plural?: string;
        };
        pie?: {
          singular?: string;
          plural?: string;
        };
        semicircle?: {
          singular?: string;
          plural?: string;
        };
      };
      notification?: {
        close?: string;
      };
      pagination?: {
        stepLabel?: string;
        summary?: string;
        summaryNoItems?: string;
      };
      rangeSelector?: {
        lower?: string;
        upper?: string;
      };
      tag?: {
        removeLabel?: {
          nameAndValue?: string;
          valueOnly?: string;
        };
      };
      select?: {
        multiple?: string;
        selected?: string;
      };
      selectMultiple?: {
        clearAll?: string;
        clearAllA11y?: string;
        open?: string;
        optionNotSelected?: string;
        optionSelected?: string;
        search?: string;
        selectAll?: string;
        selectAllA11y?: string;
        selectDrop?: string;
        selected?: string;
        selectedOfTotal?: string;
        selectedOptions?: string;
        showMore?: string;
        summarizedValue?: string;
      };
      skipLinks?: {
        skipTo?: string;
      };
      tabs?: {
        tabContents?: string;
      };
      textInput?: {
        enterSelect?: string;
        suggestionsCount?: string;
        suggestionsExist?: string;
        suggestionIsOpen?: string;
      };
      video?: {
        audioDescriptions?: string;
        captions?: string;
        closeMenu?: string;
        description?: string;
        fullScreen?: string;
        progressMeter?: string;
        scrubber?: string;
        openMenu?: string;
        pauseButton?: string;
        playButton?: string;
        volumeDown?: string;
        volumeUp?: string;
      };
    };
    format?: (...args: any[]) => void;
  };
  onAnalytics?: (data: any) => void;
  plain?: boolean;
  theme?: ThemeType;
  themeMode?: 'dark' | 'light' | 'auto';
  userAgent?: string;
}

export interface GrommetExtendedProps
  extends GrommetProps,
    Omit<
      React.DetailedHTMLProps<
        React.HTMLAttributes<HTMLDivElement>,
        HTMLDivElement
      >,
      'dir'
    > {}

declare const Grommet: React.FC<GrommetExtendedProps>;

export { Grommet };
