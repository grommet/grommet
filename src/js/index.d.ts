export * from './components/Accordion';
export * from './components/AccordionPanel';
export * from './components/Anchor';
export * from './components/Avatar';
export * from './components/Box';
export * from './components/Button';
export * from './components/Calendar';
export * from './components/Carousel';
export * from './components/Card';
export * from './components/CardBody';
export * from './components/CardFooter';
export * from './components/CardHeader';
export * from './components/Cards';
export * from './components/Chart';
export * from './components/CheckBox';
export * from './components/CheckBoxGroup';
export * from './components/Clock';
export * from './components/Collapsible';
export * from './components/Data';
export * from './components/DataChart';
export * from './components/DataFilter';
export * from './components/DataFilters';
export * from './components/DataSearch';
export * from './components/DataSort';
export * from './components/DataSummary';
export * from './components/DataTable';
export * from './components/DataTableColumns';
export * from './components/DataTableGroupBy';
export * from './components/DataView';
export * from './components/DateInput';
export * from './components/Diagram';
export * from './components/Distribution';
export * from './components/Drop';
export * from './components/DropButton';
export * from './components/FileInput';
export * from './components/Footer';
export * from './components/Form';
export * from './components/FormField';
export * from './components/Grid';
export * from './components/Grommet';
export * from './components/Header';
export * from './components/Heading';
export * from './components/Image';
export * from './components/InfiniteScroll';
export * from './components/Keyboard';
export * from './components/Layer';
export * from './components/List';
export * from './components/Main';
export * from './components/Markdown';
export * from './components/MaskedInput';
export * from './components/Menu';
export * from './components/Meter';
export * from './components/Page';
export * from './components/PageContent';
export * from './components/PageHeader';
export * from './components/Pagination';
export * from './components/Paragraph';
export * from './components/NameValueList';
export * from './components/NameValuePair';
export * from './components/Nav';
export * from './components/Notification';
export * from './components/RadioButton';
export * from './components/RadioButtonGroup';
export * from './components/RangeInput';
export * from './components/RangeSelector';
export * from './components/RoutedAnchor';
export * from './components/RoutedButton';
export * from './components/Select';
export * from './components/SelectMultiple';
export * from './components/Sidebar';
export * from './components/Skeleton';
export * from './components/SkipLink';
export * from './components/SkipLinks';
export * from './components/SkipLinkTarget';
export * from './components/Spinner';
export * from './components/Stack';
export * from './components/StarRating';
export * from './components/Table';
export * from './components/TableBody';
export * from './components/TableCell';
export * from './components/TableFooter';
export * from './components/TableHeader';
export * from './components/TableRow';
export * from './components/Tab';
export * from './components/Tabs';
export * from './components/Tag';
export * from './components/Text';
export * from './components/TextArea';
export * from './components/TextInput';
export * from './components/ThumbsRating';
export * from './components/Tip';
export * from './components/ToggleGroup';
export * from './components/Toolbar';
export * from './components/Video';
export * from './components/WorldMap';
export * from './contexts/DataContext';
export * from './contexts/AnnounceContext';
export * from './contexts/ContainerTargetContext';
export * from './contexts/ResponsiveContext';
export * from './contexts/ThemeContext';
export * from './themes';
export { defaultProps, extendDefaultTheme } from './default-props';
