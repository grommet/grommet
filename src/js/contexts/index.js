export * from './AnalyticsContext';
export * from './AnnounceContext';
export * from './ContainerTargetContext';
export * from './DataContext';
export * from './ResponsiveContext';
export * from './ResponsiveContainerContext';
export * from './RootsContext';
export * from './ThemeContext';
