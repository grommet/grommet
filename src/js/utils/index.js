export * from './animation';
export * from './mixins';
export * from './background';
export * from './border';
export * from './colors';
export * from './dates';
export * from './DOM';
export * from './graphics';
export * from './icon';
export * from './styles';
export * from './object';
export * from './pagination';
export * from './PortalContext';
export * from './readOnly';
export * from './refs';
export * from './responsive';
export * from './use-keyboard';
export * from './useId';
