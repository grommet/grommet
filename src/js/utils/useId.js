import React, { useState } from 'react';

let currentId = 0;

const getId = () => {
  // eslint-disable-next-line no-plusplus
  const id = currentId++;
  return `:r${id.toString(32)}:`;
};

const useIdGrommet = () => {
  const [id] = useState(getId);
  return id;
};

// Polyfill React 18's useId for compatibility with React 16 and 17
const useId = React.useId ?? useIdGrommet;

export { useId };
