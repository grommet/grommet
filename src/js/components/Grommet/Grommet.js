import React, { forwardRef, useEffect, useMemo, useRef, useState } from 'react';
import { createGlobalStyle } from 'styled-components';

import {
  ContainerTargetContext,
  ResponsiveContext,
  ThemeContext,
} from '../../contexts';

import {
  deepMerge,
  backgroundIsDark,
  deviceResponsive,
  getBreakpoint,
  normalizeColor,
  useForwardedRef,
} from '../../utils';
import { base as baseTheme } from '../../themes';
import { StyledGrommet } from './StyledGrommet';
import { RootsContext } from '../../contexts/RootsContext';
import { OptionsContext } from '../../contexts/OptionsContext';
import { format, MessageContext } from '../../contexts/MessageContext';
import defaultMessages from '../../languages/default.json';
import { GrommetPropTypes } from './propTypes';
import { AnalyticsProvider } from '../../contexts/AnalyticsContext';

const FullGlobalStyle = createGlobalStyle`
  body { margin: 0; }
`;

const defaultOptions = {};

const Grommet = forwardRef((props, ref) => {
  const {
    children,
    full,
    containerTarget = typeof document === 'object' ? document.body : undefined,
    theme: themeProp,
    options = defaultOptions,
    messages: messagesProp,
    onAnalytics,
    ...rest
  } = props;
  const { background, dir, themeMode, userAgent } = props;
  const [stateResponsive, setResponsive] = useState();

  const theme = useMemo(() => {
    const nextTheme = deepMerge(baseTheme, themeProp || {});

    // if user provides specific menu alignment, we don't want
    // the defaults to be included at all (can cause issues with controlMirror)
    // override merged value with themeProp value
    if (
      themeProp &&
      themeProp.menu &&
      themeProp.menu.drop &&
      themeProp.menu.drop.align
    ) {
      delete nextTheme.menu.drop.align;
      nextTheme.menu.drop.align = themeProp.menu.drop.align;
    }
    const {
      colors: { background: themeBackground },
    } = nextTheme.global;

    nextTheme.dark = (themeMode || nextTheme.defaultMode) === 'dark';

    if (
      themeMode === 'auto' &&
      typeof window !== 'undefined' &&
      window.matchMedia &&
      window.matchMedia('(prefers-color-scheme: dark)').matches
    ) {
      nextTheme.dark = true;
    }

    const color = normalizeColor(background || themeBackground, nextTheme);
    nextTheme.dark = backgroundIsDark(color, nextTheme);
    nextTheme.baseBackground = background || themeBackground;
    // This allows DataTable to intelligently set the background of a pinned
    // header or footer.
    nextTheme.background = nextTheme.baseBackground;

    if (dir) {
      nextTheme.dir = dir;
    }

    return nextTheme;
  }, [background, dir, themeMode, themeProp]);

  const messages = useMemo(() => {
    // combine the passed in messages, if any, with the default
    // messages and format function.
    const nextMessages = deepMerge(
      defaultMessages,
      messagesProp?.messages || {},
    );
    return {
      messages: nextMessages,
      format: (opts) => {
        const message = messagesProp?.format && messagesProp.format(opts);
        return typeof message !== 'undefined'
          ? message
          : format(opts, nextMessages);
      },
    };
  }, [messagesProp]);

  useEffect(() => {
    const onResize = () => {
      setResponsive(getBreakpoint(document.body.clientWidth, theme));
    };
    window.addEventListener('resize', onResize);
    onResize();
    return () => {
      window.removeEventListener('resize', onResize);
    };
  }, [theme]);

  const responsive =
    stateResponsive ||
    deviceResponsive(userAgent, theme) ||
    theme.global.deviceBreakpoints.tablet;

  const grommetRef = useForwardedRef(ref);

  // track open FocusedContainers in a global array to manage
  // focus event listeners for trapFocus
  const roots = useRef([]);
  useEffect(() => {
    if (grommetRef.current) roots.current.push(grommetRef.current);
  }, [grommetRef]);
  const rootsContextValue = useMemo(() => ({ roots }), []);

  return (
    <ThemeContext.Provider value={theme}>
      <ResponsiveContext.Provider value={responsive}>
        <RootsContext.Provider value={rootsContextValue}>
          <ContainerTargetContext.Provider value={containerTarget}>
            <OptionsContext.Provider value={options}>
              <MessageContext.Provider value={messages}>
                <AnalyticsProvider onAnalytics={onAnalytics}>
                  <StyledGrommet full={full} {...rest} ref={grommetRef}>
                    {children}
                  </StyledGrommet>
                  {full && <FullGlobalStyle />}
                </AnalyticsProvider>
              </MessageContext.Provider>
            </OptionsContext.Provider>
          </ContainerTargetContext.Provider>
        </RootsContext.Provider>
      </ResponsiveContext.Provider>
    </ThemeContext.Provider>
  );
});

Grommet.displayName = 'Grommet';
Grommet.propTypes = GrommetPropTypes;

export { Grommet };
