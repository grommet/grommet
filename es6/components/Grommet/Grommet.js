var _excluded = ["children", "full", "containerTarget", "theme", "options", "messages", "onAnalytics"];
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
function _objectWithoutPropertiesLoose(r, e) { if (null == r) return {}; var t = {}; for (var n in r) if ({}.hasOwnProperty.call(r, n)) { if (-1 !== e.indexOf(n)) continue; t[n] = r[n]; } return t; }
import React, { forwardRef, useEffect, useMemo, useRef, useState } from 'react';
import { createGlobalStyle } from 'styled-components';
import { ContainerTargetContext, ResponsiveContext, ThemeContext } from '../../contexts';
import { deepMerge, backgroundIsDark, deviceResponsive, getBreakpoint, normalizeColor, useForwardedRef } from '../../utils';
import { base as baseTheme } from '../../themes';
import { StyledGrommet } from './StyledGrommet';
import { RootsContext } from '../../contexts/RootsContext';
import { OptionsContext } from '../../contexts/OptionsContext';
import { format as _format, MessageContext } from '../../contexts/MessageContext';
import defaultMessages from '../../languages/default.json';
import { GrommetPropTypes } from './propTypes';
import { AnalyticsProvider } from '../../contexts/AnalyticsContext';
var FullGlobalStyle = createGlobalStyle(["body{margin:0;}"]);
var defaultOptions = {};
var Grommet = /*#__PURE__*/forwardRef(function (props, ref) {
  var children = props.children,
    full = props.full,
    _props$containerTarge = props.containerTarget,
    containerTarget = _props$containerTarge === void 0 ? typeof document === 'object' ? document.body : undefined : _props$containerTarge,
    themeProp = props.theme,
    _props$options = props.options,
    options = _props$options === void 0 ? defaultOptions : _props$options,
    messagesProp = props.messages,
    onAnalytics = props.onAnalytics,
    rest = _objectWithoutPropertiesLoose(props, _excluded);
  var background = props.background,
    dir = props.dir,
    themeMode = props.themeMode,
    userAgent = props.userAgent;
  var _useState = useState(),
    stateResponsive = _useState[0],
    setResponsive = _useState[1];
  var theme = useMemo(function () {
    var nextTheme = deepMerge(baseTheme, themeProp || {});

    // if user provides specific menu alignment, we don't want
    // the defaults to be included at all (can cause issues with controlMirror)
    // override merged value with themeProp value
    if (themeProp && themeProp.menu && themeProp.menu.drop && themeProp.menu.drop.align) {
      delete nextTheme.menu.drop.align;
      nextTheme.menu.drop.align = themeProp.menu.drop.align;
    }
    var themeBackground = nextTheme.global.colors.background;
    nextTheme.dark = (themeMode || nextTheme.defaultMode) === 'dark';
    if (themeMode === 'auto' && typeof window !== 'undefined' && window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches) {
      nextTheme.dark = true;
    }
    var color = normalizeColor(background || themeBackground, nextTheme);
    nextTheme.dark = backgroundIsDark(color, nextTheme);
    nextTheme.baseBackground = background || themeBackground;
    // This allows DataTable to intelligently set the background of a pinned
    // header or footer.
    nextTheme.background = nextTheme.baseBackground;
    if (dir) {
      nextTheme.dir = dir;
    }
    return nextTheme;
  }, [background, dir, themeMode, themeProp]);
  var messages = useMemo(function () {
    // combine the passed in messages, if any, with the default
    // messages and format function.
    var nextMessages = deepMerge(defaultMessages, (messagesProp == null ? void 0 : messagesProp.messages) || {});
    return {
      messages: nextMessages,
      format: function format(opts) {
        var message = (messagesProp == null ? void 0 : messagesProp.format) && messagesProp.format(opts);
        return typeof message !== 'undefined' ? message : _format(opts, nextMessages);
      }
    };
  }, [messagesProp]);
  useEffect(function () {
    var onResize = function onResize() {
      setResponsive(getBreakpoint(document.body.clientWidth, theme));
    };
    window.addEventListener('resize', onResize);
    onResize();
    return function () {
      window.removeEventListener('resize', onResize);
    };
  }, [theme]);
  var responsive = stateResponsive || deviceResponsive(userAgent, theme) || theme.global.deviceBreakpoints.tablet;
  var grommetRef = useForwardedRef(ref);

  // track open FocusedContainers in a global array to manage
  // focus event listeners for trapFocus
  var roots = useRef([]);
  useEffect(function () {
    if (grommetRef.current) roots.current.push(grommetRef.current);
  }, [grommetRef]);
  var rootsContextValue = useMemo(function () {
    return {
      roots: roots
    };
  }, []);
  return /*#__PURE__*/React.createElement(ThemeContext.Provider, {
    value: theme
  }, /*#__PURE__*/React.createElement(ResponsiveContext.Provider, {
    value: responsive
  }, /*#__PURE__*/React.createElement(RootsContext.Provider, {
    value: rootsContextValue
  }, /*#__PURE__*/React.createElement(ContainerTargetContext.Provider, {
    value: containerTarget
  }, /*#__PURE__*/React.createElement(OptionsContext.Provider, {
    value: options
  }, /*#__PURE__*/React.createElement(MessageContext.Provider, {
    value: messages
  }, /*#__PURE__*/React.createElement(AnalyticsProvider, {
    onAnalytics: onAnalytics
  }, /*#__PURE__*/React.createElement(StyledGrommet, _extends({
    full: full
  }, rest, {
    ref: grommetRef
  }), children), full && /*#__PURE__*/React.createElement(FullGlobalStyle, null))))))));
});
Grommet.displayName = 'Grommet';
Grommet.propTypes = GrommetPropTypes;
export { Grommet };