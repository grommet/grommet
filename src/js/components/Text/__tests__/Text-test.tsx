import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import { axe } from 'jest-axe';
import 'jest-styled-components';
import 'jest-axe/extend-expect';
import 'regenerator-runtime/runtime';

import * as UseIdModule from '../../../utils/useId.js';
import { Grommet } from '../../Grommet';
import { Text } from '..';

test('should have no accessibility violations', async () => {
  const { container } = render(
    <Grommet>
      <Text a11yTitle="test"> Example</Text>
    </Grommet>,
  );

  const results = await axe(container);
  expect(results).toHaveNoViolations();
  expect(container.firstChild).toMatchSnapshot();
});

test('renders', () => {
  const { container } = render(
    <Grommet>
      <Text>text</Text>
    </Grommet>,
  );

  expect(container.firstChild).toMatchSnapshot();
});

test('renders outside grommet wrapper', () => {
  const { container } = render(<Text>text</Text>);

  expect(container.firstChild).toMatchSnapshot();
});

test('accepts ref', () => {
  const ref = React.createRef<HTMLParagraphElement>();
  const { container } = render(
    <Grommet>
      <Text ref={ref}>text</Text>
    </Grommet>,
  );

  expect(ref.current).not.toBeNull();
  expect(container.firstChild).toMatchSnapshot();
});

test('renders size', () => {
  const { container } = render(
    <Grommet>
      <Text size="xsmall" />
      <Text size="small" />
      <Text size="medium" />
      <Text size="large" />
      <Text size="xlarge" />
      <Text size="xxlarge" />
      <Text size="2xl" />
      <Text size="3xl" />
      <Text size="4xl" />
      <Text size="5xl" />
      <Text size="6xl" />
    </Grommet>,
  );

  expect(container.firstChild).toMatchSnapshot();
});

test('renders textAlign', () => {
  const { container } = render(
    <Grommet>
      <Text textAlign="start" />
      <Text textAlign="center" />
      <Text textAlign="end" />
      <Text textAlign="justify" />
    </Grommet>,
  );

  expect(container.firstChild).toMatchSnapshot();
});

test('renders margin', () => {
  const { container } = render(
    <Grommet>
      <Text margin="small" />
      <Text margin="medium" />
      <Text margin="large" />
      <Text margin="none" />
      <Text margin={{ vertical: 'small' }} />
      <Text margin={{ horizontal: 'small' }} />
      <Text margin={{ bottom: 'small' }} />
      <Text margin={{ top: 'small' }} />
      <Text margin={{ left: 'small' }} />
      <Text margin={{ right: 'small' }} />
    </Grommet>,
  );

  expect(container.firstChild).toMatchSnapshot();
});

const LONG = 'a b c d e f g h i j k l m n o p q r s t u v w x y z';

test('renders truncate', () => {
  const { container } = render(
    <Grommet>
      <Text truncate={false}>{LONG}</Text>
      <Text truncate>{LONG}</Text>
    </Grommet>,
  );

  expect(container.firstChild).toMatchSnapshot();
});

test('renders color', () => {
  const { container } = render(
    <Grommet>
      <Text color="status-critical" />
    </Grommet>,
  );

  expect(container.firstChild).toMatchSnapshot();
});

test('renders tag', () => {
  const { container } = render(
    <Grommet>
      <Text as="div" />
    </Grommet>,
  );

  expect(container.firstChild).toMatchSnapshot();
});

test('proxies tag', () => {
  const { container: tagComponent } = render(
    <Grommet>
      <Text tag="div" />
    </Grommet>,
  );
  const { container: asComponent } = render(
    <Grommet>
      <Text as="div" />
    </Grommet>,
  );

  expect(tagComponent).toEqual(asComponent);
});

test('renders weight', () => {
  const { container } = render(
    <Grommet>
      <Text weight="normal" />
      <Text weight="bold" />
    </Grommet>,
  );

  expect(container.firstChild).toMatchSnapshot();
});

describe('Text component with tip', () => {
  test('renders tip on hover', () => {
    jest.spyOn(UseIdModule, 'useId').mockReturnValue('fixed-id-123');
    const { container, getByText } = render(
      <Grommet>
        <Text tip="tooltip">Default Tip</Text>
      </Grommet>,
    );

    fireEvent.mouseOver(getByText('Default Tip'));
    expect(container.firstChild).toMatchSnapshot();
    (UseIdModule.useId as jest.Mock).mockRestore();
  });
});

test('should apply a11yTitle or aria-label', () => {
  const { container, getByLabelText } = render(
    <Grommet>
      <Text a11yTitle="test"> Example</Text>
      <Text aria-label="test-2">Example</Text>
    </Grommet>,
  );

  expect(getByLabelText('test')).toBeTruthy();
  expect(getByLabelText('test-2')).toBeTruthy();
  expect(container.firstChild).toMatchSnapshot();
});
