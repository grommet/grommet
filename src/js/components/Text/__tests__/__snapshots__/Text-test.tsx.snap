// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`accepts ref 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-size: 18px;
  line-height: 24px;
}

<div
  class="c0"
>
  <span
    class="c1"
  >
    text
  </span>
</div>
`;

exports[`renders 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-size: 18px;
  line-height: 24px;
}

<div
  class="c0"
>
  <span
    class="c1"
  >
    text
  </span>
</div>
`;

exports[`renders color 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-size: 18px;
  line-height: 24px;
  color: #EB0000;
}

<div
  class="c0"
>
  <span
    class="c1"
  />
</div>
`;

exports[`renders margin 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  margin: 12px;
  font-size: 18px;
  line-height: 24px;
}

.c2 {
  margin: 24px;
  font-size: 18px;
  line-height: 24px;
}

.c3 {
  margin: 48px;
  font-size: 18px;
  line-height: 24px;
}

.c4 {
  margin: 0px;
  font-size: 18px;
  line-height: 24px;
}

.c5 {
  margin-top: 12px;
  margin-bottom: 12px;
  font-size: 18px;
  line-height: 24px;
}

.c6 {
  margin-left: 12px;
  margin-right: 12px;
  font-size: 18px;
  line-height: 24px;
}

.c7 {
  margin-bottom: 12px;
  font-size: 18px;
  line-height: 24px;
}

.c8 {
  margin-top: 12px;
  font-size: 18px;
  line-height: 24px;
}

.c9 {
  margin-left: 12px;
  font-size: 18px;
  line-height: 24px;
}

.c10 {
  margin-right: 12px;
  font-size: 18px;
  line-height: 24px;
}

<div
  class="c0"
>
  <span
    class="c1"
  />
  <span
    class="c2"
  />
  <span
    class="c3"
  />
  <span
    class="c4"
  />
  <span
    class="c5"
  />
  <span
    class="c6"
  />
  <span
    class="c7"
  />
  <span
    class="c8"
  />
  <span
    class="c9"
  />
  <span
    class="c10"
  />
</div>
`;

exports[`renders outside grommet wrapper 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
}

<span
  class="c0"
>
  text
</span>
`;

exports[`renders size 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-size: 12px;
  line-height: 18px;
}

.c2 {
  font-size: 14px;
  line-height: 20px;
}

.c3 {
  font-size: 18px;
  line-height: 24px;
}

.c4 {
  font-size: 22px;
  line-height: 28px;
}

.c5 {
  font-size: 26px;
  line-height: 32px;
}

.c6 {
  font-size: 34px;
  line-height: 40px;
}

.c7 {
  font-size: 42px;
  line-height: 48px;
}

.c8 {
  font-size: 54px;
  line-height: 60px;
}

.c9 {
  font-size: 70px;
  line-height: 76px;
}

.c10 {
  font-size: 90px;
  line-height: 96px;
}

<div
  class="c0"
>
  <span
    class="c1"
  />
  <span
    class="c2"
  />
  <span
    class="c3"
  />
  <span
    class="c4"
  />
  <span
    class="c5"
  />
  <span
    class="c6"
  />
  <span
    class="c6"
  />
  <span
    class="c7"
  />
  <span
    class="c8"
  />
  <span
    class="c9"
  />
  <span
    class="c10"
  />
</div>
`;

exports[`renders tag 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-size: 18px;
  line-height: 24px;
}

<div
  class="c0"
>
  <div
    class="c1"
  />
</div>
`;

exports[`renders textAlign 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-size: 18px;
  line-height: 24px;
  text-align: left;
}

.c2 {
  font-size: 18px;
  line-height: 24px;
  text-align: center;
}

.c3 {
  font-size: 18px;
  line-height: 24px;
  text-align: right;
}

.c4 {
  font-size: 18px;
  line-height: 24px;
  text-align: justify;
}

<div
  class="c0"
>
  <span
    class="c1"
  />
  <span
    class="c2"
  />
  <span
    class="c3"
  />
  <span
    class="c4"
  />
</div>
`;

exports[`renders tip 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-size: 18px;
  line-height: 24px;
}

@media only screen and (max-width: 768px) {

}

@media only screen and (max-width: 768px) {

}

@media only screen and (max-width: 768px) {

}

@media only screen and (max-width: 768px) {

}

@media only screen and (max-width: 768px) {

}

@media screen and (-ms-high-contrast:active),(-ms-high-contrast:none) {

}

<div
  class="c0"
>
  <span
    aria-label="tooltip"
    class="c1"
  >
    Default Tip
  </span>
</div>
`;

exports[`renders truncate 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-size: 18px;
  line-height: 24px;
}

.c2 {
  font-size: 18px;
  line-height: 24px;
  white-space: nowrap;
  max-width: 100%;
  overflow: hidden;
  text-overflow: ellipsis;
}

<div
  class="c0"
>
  <span
    class="c1"
  >
    a b c d e f g h i j k l m n o p q r s t u v w x y z
  </span>
  <span
    class="c2"
  >
    a b c d e f g h i j k l m n o p q r s t u v w x y z
  </span>
</div>
`;

exports[`renders weight 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-size: 18px;
  line-height: 24px;
  font-weight: normal;
}

.c2 {
  font-size: 18px;
  line-height: 24px;
  font-weight: bold;
}

<div
  class="c0"
>
  <span
    class="c1"
  />
  <span
    class="c2"
  />
</div>
`;

exports[`should apply a11yTitle or aria-label 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-size: 18px;
  line-height: 24px;
}

<div
  class="c0"
>
  <span
    aria-label="test"
    class="c1"
  >
     Example
  </span>
  <span
    aria-label="test-2"
    class="c1"
  >
    Example
  </span>
</div>
`;

exports[`should have no accessibility violations 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-size: 18px;
  line-height: 24px;
}

<div
  class="c0"
>
  <span
    aria-label="test"
    class="c1"
  >
     Example
  </span>
</div>
`;
